fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -718,6 +730,10 @@ public IndexOperationResult performIndexOperation(@Nonnull String indexName, @No
         final long startTime = System.nanoTime();
         CompletableFuture<FDBStoredRecord<M>> result = loadRawRecordAsync(primaryKey, sizeInfo, snapshot)
                 .thenCompose(rawRecord -> {
+                    final ByteScanLimiter byteScanLimiter = executeState.getByteScanLimiter();
+                    if (byteScanLimiter != null) {
+                        byteScanLimiter.registerScannedBytes(sizeInfo.getKeySize() + sizeInfo.getValueSize());
+                    }
```

Comment:
There's probably an argument to be made that this should update the record scan limiter as well, but I think that leaving the behavior the same at least for now is the right call.

Reply:
Yeah, I agree about not doing it here.